src/rnp/rnp.c

```diff
@@ -48,23 +48,27 @@
 #include <fcntl.h>
 #include <stdbool.h>
 #include <rnp/rnp_obsolete_defs.h>
+#include "rnpcfg.h"
 #include <rekey/rnp_key_store.h>
 #include <repgp/repgp.h>
+#include <librepgp/stream-armour.h>
 
 #include "hash.h"
 
 extern char *__progname;
 
 static const char *usage = "--help OR\n"
                            "\t--encrypt [--output=file] [options] files... OR\n"
-                           "\t--decrypt [--output=file] [options] files... OR\n\n"
+                           "\t--decrypt [--output=file] [options] files... OR\n"
                            "\t--sign [--detach] [--hash=alg] [--output=file]\n"
                            "\t\t[options] files... OR\n"
                            "\t--verify [options] files... OR\n"
                            "\t--cat [--output=file] [options] files... OR\n"
                            "\t--clearsign [--output=file] [options] files... OR\n"
                            "\t--list-packets [options] OR\n"
                            "\t--dearmor [--output=file] file OR\n"
+                           "\t--enarmor=<msg|pubkey|seckey|sign|text> \n"
```

Comment:
<msg|pubkey|seckey|sign|text> looks different from other commands description.
Also, 'text' means cleartext then it is not armoring at all.
Also --enarmor is a command, not the option - it would be more logical to make command '--enarmor' with additional option '--header', which could be 'PUBLIC KEY', 'SECRET KEY', or anything else. By default it should be 'MESSAGE'.

Reply:
Yes, other commands do not take optional argument. Here the data type is not an option - it is required. 
I actually had implementation with '--header', but it doesn't seem to be good idea. User needs to provide data type, it's not an option. Also I don't think default is good idea - we can't guess what kind of data user tries to convert in this case. Instead, we need to force user to decide.

Reply:
Forcing user to do something doesn't seem to be a good idea. We can just peek first few bytes, check the packet type, and make correct decision about the data contents. And leave '--header' option for the user in case he wants more precise control over the process. Anyway, this command is not critical for RNP functionality.

Reply:
Ah, you mean rnp should find out by itself type of the data or let the user decide. In such, yes, it would be good improvement to this. It's just not implemented now, as well, it indeed is not critical functionality (so no reason to complicate a code).

Reply:
Just my opinion but I feel that separating the options into `--enarmor` and `--header` would be rather confusing to the user. They are intrinsically linked and that is crystal clear when they are coupled in `--enarmor=msg`.

I haven't looked at the difficulty of guessing the format, but personally the way I would probably do this is I would have `--enarmor` try to guess (or force a default even), but allow overrides like `--enarmor=msg` to force a type as well.

Reply:
@flowher From rfc4880:
```
   BEGIN PGP MESSAGE
       Used for signed, encrypted, or compressed files.

   BEGIN PGP PUBLIC KEY BLOCK
       Used for armoring public keys.

   BEGIN PGP PRIVATE KEY BLOCK
       Used for armoring private keys.

   BEGIN PGP SIGNATURE
       Used for detached signatures, OpenPGP/MIME signatures, and
       cleartext signatures.  Note that PGP 2.x uses BEGIN PGP MESSAGE
       for detached signatures.
```

So basically you should just check first byte of the stream, use get_packet_type() from `stream-parse.c`, and for tag 2 use 'SIGNATURE', tag 5 - 'SECRET KEY', tag 6 - 'PUBLIC KEY', and 'MESSAGE' for all other tags. If you don't have time for this now - I may do it later on, just add an issue.

Also probably later we may wish to add `--base64` or `--raw` modifier, like gnupg does, to just encode/decode to raw base64, without headers and crc.

Reply:
ok, thanks. I would say, let's keep it stupid simple until we really needed.

Reply:
Just FYI: GnuPG wraps in "-----BEGIN PGP ARMORED FILE----" when called with "--enarmor" (which kind of sucks)